packages/trackerless-network/src/NetworkStack.ts

```diff
@@ -0,0 +1,149 @@
+/* eslint-disable @typescript-eslint/parameter-properties */
+
+import { ConnectionManager, DhtNode, DhtNodeOptions, isSamePeerDescriptor, PeerDescriptor } from '@streamr/dht'
+import { StreamrNode, StreamrNodeConfig } from './logic/StreamrNode'
+import { MetricsContext, waitForEvent3 } from '@streamr/utils'
+import { EventEmitter } from 'eventemitter3'
+import { StreamPartID } from '@streamr/protocol'
+
+interface ReadynessEvents {
+    done: () => void
+}
+
+class ReadynessListener {
+    private emitter = new EventEmitter<ReadynessEvents>()
+
+    constructor(private networkStack: NetworkStack,
+        private dhtNode: DhtNode) {
+
+        networkStack.on('stopped', this.onStopped)
+        dhtNode.on('connected', this.onConnected)
+    }
+
+    private onConnected = (_peerDescriptor: PeerDescriptor) => {
+        this.networkStack.off('stopped', this.onStopped)
+        this.dhtNode.off('connected', this.onConnected)
+        this.emitter.emit('done')
+    }
+
+    private onStopped = () => {
+        this.networkStack.off('stopped', this.onStopped)
+        this.dhtNode.off('connected', this.onConnected)
+        this.emitter.emit('done')
+    }
+
+    public async waitUntilReady(timeout: number): Promise<void> {
+        if (this.dhtNode.getNumberOfConnections() > 0) {
+            return
+        } else {
+            await waitForEvent3<ReadynessEvents>(this.emitter, 'done', timeout)
+
+        }
+    }
+}
+
+export interface NetworkOptions {
+    layer0: DhtNodeOptions
+    networkNode: StreamrNodeConfig
+    metricsContext?: MetricsContext
```

Comment:
There is a `metricsContext` field also inside `StreamrNodeConfig`. Most likely both of the field have the same value? Could this interface just just `networkNode.metricsContext` instead of introducing a separate field?